### ISSUE TITLE
Questions about ModernNCA

### ISSUE THREAD

**Issue by DYWLK** [qile2000/LAMDA-TALENT#13]
Very clear and comprehensive work of ModernNCA! Respect to the team.
However, I have some questions about learning objective.
IIUC, during training, all of samples in candidate_x are necessary to be part of backward propagation. Thus, no matter what batch_size is set, the actual "batch_size" is the number of samples in candidate_x.
Is that right?

**Comment by Yinhuaihong** [comment]
Hi, thank you for your interest in our work! Let me address your question:

>"All the samples in candidate_x are necessary to be part of backward propagation."

That's correct, your understanding is accurate.

>"The actual 'batch_size' is the number of samples in candidate_x."

This depends on how you define "batch size."

From the perspective of gradient backpropagation, both x (mini-batch samples) and candidate_x (sampled neighbors) are included in the same batch for the gradient update. However, from the perspective of the retrieval model, candidate_x serves as the support set, and x is the query set. We still train using a mini-batch approach, where samples (x) are randomly selected from the training dataset as a batch. Once all the training samples have been used as query sets for gradient backpropagation, an epoch is considered complete.
